jans-cli-tui/setup.py

```
@@ -6,10 +6,31 @@
 from setuptools import setup
 from setuptools import find_packages
 from setuptools.command.install import install
 from urllib.request import urlretrieve
+from urllib.error import HTTPError, URLError
+
+def _download_with_retry(url, dest, retries=5, delay=10, timeout=60) -> None:
+    for attempt in range(retries):
+        prev_timeout = socket.getdefaulttimeout()
+        try:
+            socket.setdefaulttimeout(timeout)
+            urlretrieve(url, dest)
```

Ruff (0.15.21)

[error] Audit URL open for permitted schemes. Allowing use of `file:` or custom schemes is often unexpected.

(S310)